### PR TITLE
Port schema from php course reviews to courses

### PR DESCRIPTION
**Pull request by alokedesai**

### CODE REVIEW

File: aspc/courses/models.py [hunk 1]
```diff
@@ -279,3 +282,27 @@ def __unicode__(self):
...
+    comments = models.TextField(blank=True, null=True)
+
+    overall_rating = models.FloatField()
+    useful_rating = models.FloatField()
+    difficulty_rating = models.FloatField()
+
+    professor_competency_rating = models.FloatField()
+    professor_lecturing_style_rating = models.FloatField()
+    professor_enthusiasm_rating = models.FloatField()
+    professor_approachable_rating = models.FloatField()
```
Comment (mattdahl) [inline aspc/courses/models.py]: Let's talk about reforming some of these rating criteria at the meeting tomorrow. I think "enthusiasm" and "lecturing style" are redundant, though I know both are holdovers from the old system. Also, what do you think about having a field for reviewers to (optionally) say what grade they got in the class?

File: aspc/courses/models.py [hunk 2]
```diff
@@ -37,6 +39,7 @@ class Meta:
 
 class Instructor(models.Model):
     name = models.CharField(max_length=100)
+    rating = models.FloatField(blank = True, null = True)
```
Comment (mattdahl) [inline aspc/courses/models.py]: When do the `Instructor.rating` and `Course.rating` fields get set? Shouldn't there be a callback in `CourseReview#save()` that sets this information for its constituent `Instructor` and `Course` objects?

**Comment by alokedesai** [comment]
@mattdahl yep, didn't realize someone made a PR about this. Will push the save stuff later tonight.

File: aspc/courses/models.py (hunk 1 above)
Comment (alokedesai) [inline aspc/courses/models.py]: IMO I think people knowing grades would have a negative effect on what classes people take (though definitely open to it if other people think so). And yeah sounds good about all the ratings, having all of them sounds a little excessive to me

**Comment by alokedesai** [comment]
@dmsm @mattdahl Can you finish/critique this? Not much work is left on this, if at all.

File: aspc/courses/models.py [hunk 3]
```diff
@@ -279,3 +287,42 @@ def __unicode__(self):
...
+    def create(self, *args, **kwargs):
+        self.update_course_and_instructor_rating()
+        super(CourseReview, self).create(*args, **kwargs)
+
+    def save(self, *args, **kwargs):
+        self.update_course_and_instructor_rating()
+        super(CourseReview, self).save(*args, **kwargs)
+
+    def delete(self, *args, **kwargs):
+        self.overall_rating = 0.0
```
Comment (mattdahl) [inline aspc/courses/models.py]: this is genius, I wouldn't have thought to do this

**Comment by mattdahl** [comment]
I rebased the migrations in this PR against those in #58, and then reopened a PR identical to this one against the `course_reviews_app` instead. This changes have been merged in #60 so I am closing this.

**Pull request closed without merging**